### DISCUSSION TITLE
Disable/enable Map interaction

### DISCUSSION THREAD

**Discussion by zacBkh** [uber/react-map-gl#2145] [Q&A] [answered]
Hello!

I am using react-map-gl in a multi-step form. Let's say the map is step 3.
When I am done with the map and movingto step 4, I want the map to be completely disabled and prevent any interaction with it.

I tried changing dynamically the option `interactive` through props like this, with `shouldBeDisabled` being a props which checks if the map should be disabled or not, depending on the current step.

```
<Map
   interactive={!shouldBeDisabled}
>
```            

The props change effectively (I can see them changing in dev tools) but the map remains interactive.
I feel it's like if this property is only working on mount, and is not mutable afterward.

Am I missing something?

**Comment by Pessimistress** [comment, marked as answer]
See [documentation](https://visgl.github.io/react-map-gl/docs/api-reference/map#other-options) about `interactive`:

> Props in this section are not reactive. They are only used once when the Map instance is constructed.

This is because the underlying Map class does not offer any API for these options to be changed.

You can turn off the [input handlers](https://visgl.github.io/react-map-gl/docs/api-reference/map#input-handler-options) to disable interaction.

**Reply by zacBkh** [reply]
Thank you so much for your quick feedback!!
